background-processor/src/lib.rs

```diff
@@ -63,40 +65,50 @@ impl BackgroundProcessor {
 	/// [`ChannelManager`]: lightning::ln::channelmanager::ChannelManager
 	/// [`ChannelManager::write`]: lightning::ln::channelmanager::ChannelManager#impl-Writeable
 	/// [`FilesystemPersister::persist_manager`]: lightning_persister::FilesystemPersister::persist_manager
-	pub fn start<PM, Signer, M, T, K, F, L>(persist_manager: PM, manager: Arc<ChannelManager<Signer, Arc<M>, Arc<T>, Arc<K>, Arc<F>, Arc<L>>>, logger: Arc<L>) -> Self
-	where Signer: 'static + Sign,
-	      M: 'static + chain::Watch<Signer>,
-	      T: 'static + BroadcasterInterface,
-	      K: 'static + KeysInterface<Signer=Signer>,
-	      F: 'static + FeeEstimator,
-	      L: 'static + Logger,
-	      PM: 'static + Send + Fn(&ChannelManager<Signer, Arc<M>, Arc<T>, Arc<K>, Arc<F>, Arc<L>>) -> Result<(), std::io::Error>,
+	pub fn start<PM, Signer, M, T, K, F, L, Descriptor: 'static + SocketDescriptor + Send, CM, RM>(
+		persist_channel_manager: PM,
+		channel_manager: Arc<ChannelManager<Signer, Arc<M>, Arc<T>, Arc<K>, Arc<F>, Arc<L>>>,
+		peer_manager: Arc<PeerManager<Descriptor, Arc<CM>, Arc<RM>, Arc<L>>>, logger: Arc<L>,
+	) -> Self
+	where
+		Signer: 'static + Sign,
+		M: 'static + chain::Watch<Signer>,
+		T: 'static + BroadcasterInterface,
+		K: 'static + KeysInterface<Signer = Signer>,
+		F: 'static + FeeEstimator,
+		L: 'static + Logger,
+		CM: 'static + ChannelMessageHandler,
+		RM: 'static + RoutingMessageHandler,
+		PM: 'static
+			+ Send
+			+ Fn(
+				&ChannelManager<Signer, Arc<M>, Arc<T>, Arc<K>, Arc<F>, Arc<L>>,
+			) -> Result<(), std::io::Error>,
 	{
 		let stop_thread = Arc::new(AtomicBool::new(false));
 		let stop_thread_clone = stop_thread.clone();
 		let handle = thread::spawn(move || -> Result<(), std::io::Error> {
 			let mut current_time = Instant::now();
 			loop {
-				let updates_available = manager.await_persistable_update_timeout(Duration::from_millis(100));
+				peer_manager.process_events();
```

Comment:
Hmm, unrelated to this PR, but we really need to fix support for knowing we need to persist if something happened while we *weren't* awaiting. Probably some unique id that just gets incremented every time we would wake up any waiters and then get that as a part of the write process. Then we can always check that the ID is the latest.